metrics/bucket.go

```diff
@@ -0,0 +1,51 @@
+package metrics
+
+import (
+	"context"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+// NewCounter creates a Float64Counter with units of bytes and a default set of aggregation
+// bounds from .25 kilobytes to 10 megabytes.
+func NewBucketCounter(name, desc string, tagKeys ...tag.Key) *Float64Counter {
+	defaultBounds := []float64{250, 500, 1000, 5000, 10000, 50000, 100000, 500000, 1000000, 5000000, 10000000}
+	return NewBucketCounterWithBuckets(name, desc, stats.UnitBytes, defaultBounds, tagKeys...)
+}
+
+// NewCounterWithBuckets creates a Float64Counter wrapping an opencensus int64 measurement.
+func NewBucketCounterWithBuckets(name, desc, unit string, bounds []float64, tagKeys ...tag.Key) *Float64Counter {
+	log.Infof("registering avg counter: %s - %s", name, desc)
+	measure := stats.Float64(name, desc, unit)
+	iView := &view.View{
+		Name:        name,
+		Measure:     measure,
+		Description: desc,
+		TagKeys:     tagKeys,
+		Aggregation: view.Distribution(bounds...),
+	}
+	if err := view.Register(iView); err != nil {
+		// a panic here indicates a developer error when creating a view.
+		// Since this method is called in init() methods, this panic when hit
+		// will cause running the program to fail immediately.
+		panic(err)
+	}
+
+	return &Float64Counter{
+		measureCt: measure,
+		view:      iView,
+	}
+}
+
+// Float64Counter wraps an opencensus float64 measure that is uses as a counter.
+type Float64Counter struct {
+	measureCt *stats.Float64Measure
+	view      *view.View
+}
+
+// Add adds `m` to the values tracked by Float64Counter
+func (c *Float64Counter) Add(ctx context.Context, m uint64) {
```

Comment:
It would be _**really**_ nice to have structured logging here (and in other metrics) it could tell us the size of each block, tipset, and message referenced by CID. In the Timer metrics, it could tell us how long each tipset, block and message took to processes. As this exists now, these metrics only tell half the story without adequate logging.